android/src/main/java/io/grpc/android/AndroidChannelBuilder.java

```diff
@@ -73,18 +73,43 @@
 
   @Nullable private Context context;
 
-  public static final AndroidChannelBuilder forTarget(String target) {
+  /**
+   * Creates a new builder with the given target string that will be resolved by
+   * {@link io.grpc.NameResolver}.
+   */
+  public static AndroidChannelBuilder forTarget(String target) {
     return new AndroidChannelBuilder(target);
   }
 
+  /**
+   * Creates a new builder with the given host and port.
+   */
   public static AndroidChannelBuilder forAddress(String name, int port) {
     return forTarget(GrpcUtil.authorityFromHostAndPort(name, port));
   }
 
+  /**
+   * Creates a new builder, which delegates to the given ManagedChannelBuilder.
+   *
+   * @deprecated Use {@link #usingBuilder(ManagedChannelBuilder)} instead.
+   */
+  @Deprecated
   public static AndroidChannelBuilder fromBuilder(ManagedChannelBuilder<?> builder) {
     return new AndroidChannelBuilder(builder);
   }
 
+  /**
+   * Creates a new builder, which delegates to the given ManagedChannelBuilder.
+   *
+   * <p>The builder being delegated to can still be modified and configuration changes will
```

Comment:
I think we should say, 'The provided {@code builder} becomes "owned" by AndoidChannelBuilder. The caller should not modify the provided builder and AndroidChannelBuilder may modify it.'

Reply:
Is that the expected usage? Later there will be no configuration APIs in `AndroidChannelBuilder` (`transportExecutor()`, `sslSocketFactory()` and `scheduledExecutorService()` will be deleted), so users would not modify `AndroidChannelBuilder` after calling `usingBuilder()` (calling `forTarget()` or `forAddress()` would result in rebuild from scratch, which is not the intended usage). I think users should still be able to modify the provided builder until the channel gets built.

Reply:
I would expect them to change those settings before passing to this builder. Is there a reason they would want to change it later?

Note that it is something we can begin allowing in the future.

Reply:
Ok, that's fair. But `AndroidChannelBuilder` won't modify the provided builder, and users could still use the provided builder, with some modification on configurations, to build another channel. Would the following usage still be legit?

```java
OkHttpChannelBuilder okhttpChBuilder = OkHttpChannelBuilder.forTarget("localhost:8080");
ManagedChannel ch1 = AndroidChannelBuilder.usingBuilder(okhttpChBuilder).build();
ManagedChannel ch2 = AndroidChannelBuilder.usingBuilder(okhttpChBuilder.transportExecutor(...)).build();
```

Reply:
As discussed offline, `AndroidChannelBuilder` should reserve the right to modify the provided builder, even though it currently does not. 

Therefore, the above usage should be discouraged as we do not guarantee `okhttpChBuilder` is not modified after building `ch1`, which may cause unexpected configuration for `ch2`.

Reply:
Done.